From 93416ab17fa6f7b33f07325ff9a4f06b9daaed02 Mon Sep 17 00:00:00 2001
From: Eric Yang <eric.yang2@amd.com>
Date: Mon, 11 Jul 2016 11:45:51 -0400
Subject: [PATCH 0616/1722] drm/amd/dal: Reduce logging spam for multiplane
 usecase

Do not print surface parameters by default.

Print timing sync info only for diags

Do not print commit_surface calls for multiplane use case except
when entering and exiting

Change-Id: I27f628a115165239bc467fd13bfbb222e329f4d0
Signed-off-by: Eric Yang <eric.yang2@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/basics/logger.c           |  4 +++-
 drivers/gpu/drm/amd/dal/dc/core/dc.c                 | 20 +++++++++++++-------
 .../gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c  | 11 +++++++----
 drivers/gpu/drm/amd/dal/include/logger_types.h       |  1 +
 4 files changed, 24 insertions(+), 12 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/basics/logger.c b/drivers/gpu/drm/amd/dal/dc/basics/logger.c
index b49d9a6..ff5da87 100644
--- a/drivers/gpu/drm/amd/dal/dc/basics/logger.c
+++ b/drivers/gpu/drm/amd/dal/dc/basics/logger.c
@@ -249,6 +249,8 @@ struct log_major_mask_info {
 
 #define LG_WARN_MSK ~(1 << LOG_MINOR_COMPONENT_TOPOLOGY_MANAGER)
 
+#define LG_IF_MSK ~(1 << LOG_MINOR_COMPONENT_SURFACE)
+
 #define LG_HW_TRACE_MSK (LG_ALL_MSK & \
 		~((1 << LOG_MINOR_HW_TRACE_LINK_TRAINING) \
 				| (1 << LOG_MINOR_HW_TRACE_AUDIO)))
@@ -257,7 +259,7 @@ static const struct log_major_mask_info log_major_mask_info_tbl[] = {
 	/* LogMajor                  major name       default     MinorTble                    tblElementCnt */
 	{{LOG_MAJOR_ERROR,           "Error"       }, LG_ALL_MSK, component_minor_info_tbl,    NUM_ELEMENTS(component_minor_info_tbl)},
 	{{LOG_MAJOR_WARNING,         "Warning"     }, LG_WARN_MSK, component_minor_info_tbl,   NUM_ELEMENTS(component_minor_info_tbl)},
-	{{LOG_MAJOR_INTERFACE_TRACE, "IfTrace"     }, LG_ALL_MSK, component_minor_info_tbl,    NUM_ELEMENTS(component_minor_info_tbl)},
+	{{LOG_MAJOR_INTERFACE_TRACE, "IfTrace"     }, LG_IF_MSK, component_minor_info_tbl,    NUM_ELEMENTS(component_minor_info_tbl)},
 	{{LOG_MAJOR_HW_TRACE,        "HwTrace"     }, LG_HW_TRACE_MSK, hw_trace_minor_info_tbl, NUM_ELEMENTS(hw_trace_minor_info_tbl)},
 	{{LOG_MAJOR_MST,             "MST"         }, LG_ALL_MSK, mst_minor_info_tbl,          NUM_ELEMENTS(mst_minor_info_tbl)},
 	{{LOG_MAJOR_DCS,             "DCS"         }, LG_ALL_MSK, dcs_minor_info_tbl,          NUM_ELEMENTS(dcs_minor_info_tbl)},
diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc.c b/drivers/gpu/drm/amd/dal/dc/core/dc.c
index fdea6e2..2e6f32d 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc.c
@@ -925,13 +925,19 @@ bool dc_commit_surfaces_to_target(
 		if (new_surfaces[i]->visible)
 			new_enabled_surface_count++;
 
-	dal_logger_write(core_dc->ctx->logger,
-				LOG_MAJOR_INTERFACE_TRACE,
-				LOG_MINOR_COMPONENT_DC,
-				"%s: commit %d surfaces to target 0x%x\n",
-				__func__,
-				new_surface_count,
-				dc_target);
+	/*
+	 * Do not print if we have 2 surfaces previously and we are currently
+	 * comiting 2 surfaces. Since the multi-plane case generate a lot of
+	 * spam
+	 */
+	if (!((new_surface_count > 1) && target_status->surface_count > 1))
+		dal_logger_write(core_dc->ctx->logger,
+					LOG_MAJOR_INTERFACE_TRACE,
+					LOG_MINOR_COMPONENT_DC,
+					"%s: commit %d surfaces to target 0x%x\n",
+					__func__,
+					new_surface_count,
+					dc_target);
 
 	if (!resource_attach_surfaces_to_context(
 			new_surfaces, new_surface_count, dc_target, context)) {
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
index 74e9328..6d3314b 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
@@ -1751,6 +1751,7 @@ static void init_hw(struct core_dc *dc)
 	}
 }
 
+#ifdef DIAGS_BUILD
 static void print_context_timing_status(
 		const struct core_dc *dc,
 		struct resource_context *res_ctx)
@@ -1781,6 +1782,7 @@ static void print_context_timing_status(
 
 	dal_logger_close(&entry);
 }
+#endif
 
 /* TODO: move this to apply_ctx_tohw some how?*/
 static void dce110_power_on_pipe_if_needed(
@@ -1877,7 +1879,7 @@ static void dce110_program_front_end_for_pipe(struct core_dc *dc,
 
 	dal_logger_write(dc->ctx->logger,
 			LOG_MAJOR_INTERFACE_TRACE,
-			LOG_MINOR_COMPONENT_DC,
+			LOG_MINOR_COMPONENT_SURFACE,
 			"Pipe:%d 0x%x: addr hi:0x%x, "
 			"addr low:0x%x, "
 			"src: %d, %d, %d,"
@@ -1902,8 +1904,8 @@ static void dce110_program_front_end_for_pipe(struct core_dc *dc,
 
 
 	dal_logger_write(dc->ctx->logger,
-			LOG_MAJOR_HW_TRACE,
-			LOG_MINOR_HW_TRACE_SET_MODE,
+			LOG_MAJOR_INTERFACE_TRACE,
+			LOG_MINOR_COMPONENT_SURFACE,
 			"Pipe %d: width, height, x, y\n"
 			"viewport:%d, %d, %d, %d\n"
 			"recout:  %d, %d, %d, %d\n",
@@ -1957,8 +1959,9 @@ static enum dc_status apply_ctx_to_surface(
 				false);
 	}
 
+#ifdef DIAGS_BUILD
 	print_context_timing_status(dc, &context->res_ctx);
-
+#endif
 	return DC_OK;
 }
 
diff --git a/drivers/gpu/drm/amd/dal/include/logger_types.h b/drivers/gpu/drm/amd/dal/include/logger_types.h
index 26960e8..ce2b649 100644
--- a/drivers/gpu/drm/amd/dal/include/logger_types.h
+++ b/drivers/gpu/drm/amd/dal/include/logger_types.h
@@ -102,6 +102,7 @@ enum log_minor {
 	LOG_MINOR_COMPONENT_ISR,
 	LOG_MINOR_COMPONENT_BIOS,
 	LOG_MINOR_COMPONENT_DC,
+	LOG_MINOR_COMPONENT_SURFACE,
 	LOG_MINOR_COMPONENT_IRQ_SERVICE,
 
 /**
-- 
2.7.4

